rpc/legacyrpc/server.go

```diff
@@ -598,3 +604,22 @@ func (s *Server) requestProcessShutdown() {
 func (s *Server) RequestProcessShutdown() <-chan struct{} {
 	return s.requestShutdownChan
 }
+
+// RequestProcessDebugLevel returns a channel that is sent to when change debug
+// level request arrives
+func (s *Server) RequestProcessDebugLevel() <-chan string {
+	return s.requestDebugLevelChan
+}
+
+// ResponseProcessDebugLevel returns response chan that can store an empty string
+// if no error is found or error description string
+func (s *Server) ResponseProcessDebugLevel() chan<- string {
+	return s.responseDebugLevelChan
+}
+
+// setRequestProcessDebugLevel sends debug level to request chan and waits for
+// response message
+func (s *Server) setRequestProcessDebugLevel(debugLevel string) string {
+	s.requestDebugLevelChan <- debugLevel
```

Comment:
pass an error channel here too. from main, send a non-nil error if the debug level spec was invalid.  you can either send a nil error, or close the channel, if the log level is successfully changed.  By doing this we prevent a logic race where two callers both calling debuglevel at the same time might not receive the correct error.

Reply:
I thought I had considered the logic race situation but seems I've not handled it fully.  Another commit is ready for review.